tiledb/sm/query/readers/result_tile.cc

```diff
@@ -220,6 +225,11 @@ ResultTile::TileTuple* ResultTile::tile_tuple(const std::string& name) {
     return &delete_timestamps_tile_.value();
   }
 
+  if (delete_condition_index_tile_.has_value() &&
```

Comment:
At the moment there are... Eventually I would agree this would be better but for now that would make this PR more extensive than it needs to be.

Reply:
I'm not sure how to word a comment and I'd like to fix properly for all fields in a separate PR (https://app.shortcut.com/tiledb-inc/story/20923/enforce-stronger-contract-in-resulttile-tile-tuple). Mostly in a lot of cases, we initialize the tuple at the top of some function in the reader, then it might never get used as the data might be coming from somewhere else (for timestamps for example, this could be coming from the fragment timestamp).